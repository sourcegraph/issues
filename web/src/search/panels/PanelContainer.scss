.panel-container {
    &__header {
        display: flex;
        flex-direction: row;
        border-bottom: 1px solid $color-border;

        .theme-light & {
            border-bottom-color: $color-light-border;
        }

        &-text {
            font-weight: 600 !important;
            font-size: 16px;
            line-height: 24px;
            letter-spacing: 0.05em;
            margin-bottom: 0.25rem;
            flex-grow: 1;
        }
    }
}
