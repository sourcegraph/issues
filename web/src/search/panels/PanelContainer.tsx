import * as React from 'react'

interface Props {
    title: string
    state: 'loading' | 'content' | 'empty'
    loadingDisplay: JSX.Element
    contentDisplay: JSX.Element
    emptyDisplay: JSX.Element
    actionButtons?: JSX.Element
    className?: string
}

export const PanelContainer: React.FunctionComponent<Props> = ({
    title,
    state,
    loadingDisplay,
    contentDisplay,
    emptyDisplay,
    actionButtons,
    className,
}) => (
    <div className={`${className || ''} panel-container`}>
        <div className="panel-container__header">
            <h3 className="panel-container__header-text">{title}</h3>
            {actionButtons}
        </div>

        {state === 'loading' && loadingDisplay}
        {state === 'content' && contentDisplay}
        {state === 'empty' && emptyDisplay}
    </div>
)
