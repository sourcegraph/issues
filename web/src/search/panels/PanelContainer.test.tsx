import renderer from 'react-test-renderer'
import React from 'react'
import { PanelContainer } from './PanelContainer'

describe('PanelContainer', () => {
    const defaultProps = {
        title: 'Test Panel',
        state: 'loading',
        loadingDisplay: <div>Loading</div>,
        contentDisplay: <div>Content</div>,
        emptyDisplay: <div>Empty</div>,
    }

    test('loading state', () => {
        expect(renderer.create(<PanelContainer {...defaultProps} state="loading" />)).toMatchSnapshot()
    })

    test('empty state', () => {
        expect(renderer.create(<PanelContainer {...defaultProps} state="empty" />)).toMatchSnapshot()
    })

    test('content state', () => {
        expect(renderer.create(<PanelContainer {...defaultProps} state="content" />)).toMatchSnapshot()
    })

    test('with action buttons', () => {
        const actionButtons = <button type="button">Button</button>
        expect(
            renderer.create(<PanelContainer {...defaultProps} state="content" actionButtons={actionButtons} />)
        ).toMatchSnapshot()
    })
})
