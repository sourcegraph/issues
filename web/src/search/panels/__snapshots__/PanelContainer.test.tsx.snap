// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelContainer content state 1`] = `
<div
  className=" panel-container"
>
  <div
    className="panel-container__header"
  >
    <h3
      className="panel-container__header-text"
    >
      Test Panel
    </h3>
  </div>
  <div>
    Content
  </div>
</div>
`;

exports[`PanelContainer empty state 1`] = `
<div
  className=" panel-container"
>
  <div
    className="panel-container__header"
  >
    <h3
      className="panel-container__header-text"
    >
      Test Panel
    </h3>
  </div>
  <div>
    Empty
  </div>
</div>
`;

exports[`PanelContainer loading state 1`] = `
<div
  className=" panel-container"
>
  <div
    className="panel-container__header"
  >
    <h3
      className="panel-container__header-text"
    >
      Test Panel
    </h3>
  </div>
  <div>
    Loading
  </div>
</div>
`;

exports[`PanelContainer with action buttons 1`] = `
<div
  className=" panel-container"
>
  <div
    className="panel-container__header"
  >
    <h3
      className="panel-container__header-text"
    >
      Test Panel
    </h3>
    <button
      type="button"
    >
      Button
    </button>
  </div>
  <div>
    Content
  </div>
</div>
`;
